### PR TITLE
web_service: Unify links for web service endpoints

### PR DESCRIPTION
**Pull request by FearlessTobi**
This PR aims to unify the links for web service endpoints into one. 
All links for the respective features like telemetry, rooms, etc. are then directly hardcoded in the code.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4138)

### CODE REVIEW

File: src/dedicated_room/citra-room.cpp [hunk 1]
```diff
@@ -166,7 +166,7 @@ int main(int argc, char** argv) {
     }
     if (announce) {
         std::cout << "Hosting a public room\n\n";
-        Settings::values.announce_multiplayer_room_endpoint_url = announce_url;
+        Settings::values.web_services_endpoint_url = announce_url;
```
Comment (zhaowenlan1779) [inline src/dedicated_room/citra-room.cpp]: minor nit: it seems that this is not the same behavior as before.
Now you should pass the `announce_url` without the endpoint.

**Comment by FearlessTobi** [comment]
The announce-url now has been renamed to make clear the functionality works differently now. If I don't forget, I'll also get in touch with the people hosting rooms to tell them about the changes (mainly AngryPotato).

File: src/dedicated_room/citra-room.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ static void PrintHelp(const char* argv0) {
                  "--preferred-game-id The preferred game-id for this room\n"
                  "--username          The username used for announce\n"
                  "--token             The token used for announce\n"
-                 "--announce-url      The url to the announce server\n"
+                 "--endpoint-url      The endpoint url to the announce server\n"
```
Comment (zhaowenlan1779) [inline src/dedicated_room/citra-room.cpp]: `endpoint` is not suitable here. You may want to use `web-services-url`, `api-url` or something.
I believe that "api.citra-emu.org" is *not* an API endpoint, while "api.citra-emu.org/whatever" is.

Actually, I think the `endpoint` in the `Settings::values.web_services_endpoint_url` should be removed as well.

Comment (chris062689) [reply src/dedicated_room/citra-room.cpp]: https://api.citra-emu.org/ does not resolve to an endpoint currently.

File: src/dedicated_room/citra-room.cpp [hunk 3]
```diff
@@ -42,7 +42,7 @@ static void PrintHelp(const char* argv0) {
                  "--preferred-game-id The preferred game-id for this room\n"
                  "--username          The username used for announce\n"
                  "--token             The token used for announce\n"
-                 "--announce-url      The url to the announce server\n"
+                 "--web-services-url  The web service url to the announce server\n"
```
Comment (zhaowenlan1779) [inline src/dedicated_room/citra-room.cpp]: This is .. hard to understand
`The web service url to the announce server`
why not just `Citra Web Services url`

**Comment by chris062689** [comment]
Web API is more accurate than web services.

**Comment by zhaowenlan1779** [comment]
@chris062689 By "web services" I think it is referring to the proper noun "Citra Web Services", which is accurate.

**Comment by chris062689** [comment]
@zhaowenlan1779 I understand, but just to give you a clear definition, as the two terms are used interchangeably on Discord.

Web API - Specifically https://api.citra-emu.org/ endpoints (which the above variable references) 

Web Services - Includes the API, Discord Bot, User Profile Webpage, Game DB Submission webpage, etc. 

So I think Web API would be a more accurate representation.

**Pull request merged**